nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/ListS3.java

```diff
@@ -267,26 +267,28 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             commit(context, session, listCount);
             listCount = 0;
         } while (bucketLister.isTruncated());
-        currentTimestamp = maxTimestamp;
+
+        if (maxTimestamp > currentTimestamp) {
+            currentTimestamp = maxTimestamp;
+        }
 
         final long listMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
         getLogger().info("Successfully listed S3 bucket {} in {} millis", new Object[]{bucket, listMillis});
 
         if (!commit(context, session, listCount)) {
-            if (currentTimestamp > 0) {
-                persistState(context);
-            }
```

Comment:
Since `currentTimestamp` is never overwritten by a `maxTimestamp` with a value of zero, this check shouldn't be necessary anymore.